addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRPort.java

```diff
@@ -0,0 +1,498 @@
+                if (seEvent.getNewValue() && !portStatus.bi) {
+                    logger.debug("Serial Communication is broken");
+
+                    p1Parser.abortTelegram();
+                    dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.LINE_BROKEN);
+                }
+                portStatus.bi = seEvent.getNewValue();
+                break;
+            case SerialPortEvent.FE:
+                logger.debug("Frame error: {}", seEvent.getNewValue());
+                if (seEvent.getNewValue() && !portStatus.fe) {
+                    if (portStatus.pe) {
+                        // Both a frame and parity error --> possible wrong baud rate
+                        logger.debug("Experienced both parity and frame error caused possibly by a wrong baudrate");
+
+                        p1Parser.abortTelegram();
+                        dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.WRONG_BAUDRATE);
+                    } else {
+                        // Handle frame error
+                        if (!lenientMode) {
+                            p1Parser.abortTelegram();
+                            dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.READ_ERROR);
+                        }
+                    }
+                }
+                portStatus.fe = seEvent.getNewValue();
+                break;
+            case SerialPortEvent.OE:
+                logger.debug("Overrun Error: {}", seEvent.getNewValue());
+                if (seEvent.getNewValue() && !portStatus.oe) {
+                    logger.debug("Experienced overrun error");
+
+                    if (!lenientMode) {
+                        p1Parser.abortTelegram();
+                        dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.READ_ERROR);
+                    }
+                }
+                portStatus.oe = seEvent.getNewValue();
+                break;
+            case SerialPortEvent.PE:
+                logger.debug("Parity error: {}", seEvent.getNewValue());
+                if (seEvent.getNewValue() && !portStatus.pe) {
+                    if (portStatus.fe) {
+                        // Both a frame and parity error --> possible wrong baud rate
+                        logger.debug("Experienced both parity and frame error caused possibly by a wrong baudrate");
+
+                        p1Parser.abortTelegram();
+                        dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.WRONG_BAUDRATE);
+                    } else {
+                        // Handle parity error
+                        if (!lenientMode) {
+                            p1Parser.abortTelegram();
+                            dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.READ_ERROR);
+                        }
+                    }
+                }
+                portStatus.pe = seEvent.getNewValue();
+                break;
+            default: /* do nothing */
```

Comment:
Very big method, maybe the cases should have their own methods with a clear name stating what they do